sillsdev/web-languageforge
# Delete account

Fixes #1362

## Description

Adding a tab to the user profile for a user to delete their account. If a user is a project owner, a message letting them know to transfer or delete their owned projects is displayed. I have noticed that every so often, on a first load of the user profile, the delete dialog is still shown to users who own projects, due to mistiming of boolean values being set, possibly due to race conditions.

### Type of Change

Only keep lines below that describe this change, then delete the rest.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update - maybe add a post to the community site
- UI change

## Screenshots
When a user owns projects (cannot yet delete their account):
![image](https://user-images.githubusercontent.com/56163492/211514824-8c8b09a9-3296-440b-86f5-c60c491c5e35.png)

When a user has no projects (can go ahead and delete their account):
![image](https://user-images.githubusercontent.com/56163492/211515147-a5b2771e-15eb-4359-9cdd-1724b28e542a.png)

## Checklist

- [] I have added tests that prove my fix is effective or that my feature works

## How to test

1. Access this tab as a user who owns projects. You should see a list of links to the projects you own, with instructions and a link to a help article about transferring ownership.
2. Access this tab as a user who does not own any projects. You should be able to fill in the DELETE box, click the Delete Account button, confirm your decision in the modal, and be redirected to the login page, with your account successfully deleted.

## qa.languageforge.org testing

Testers should add his/her findings to end of the PR in a comment and include screenshots, files, etc that are beneficial.
